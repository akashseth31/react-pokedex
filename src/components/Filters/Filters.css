.filters {
    position: relative;
}

.filter-header {
    border: 1px solid;
    padding: 3px 7px;
    font-size: 16px;
    width: 100%;
    outline: none;
    text-align: left;
    background-image: url(../../helpers/assets/caret-down.png);
    background-size: auto 100%;
    background-repeat: no-repeat;
    background-position: right;
}

.filter-body {
    position: absolute;
    bottom: -5px;
    left: 0;
    right: 0;
    border: 1px solid;
    max-height: 300px;
    overflow-y: auto;
    transform: translateY(100%);
    background: #ffffff;
    padding: 15px;
    z-index: 1;
    box-shadow: 3px 3px 3px 3px #ccc;
}

.filter-item {
    border: 1px solid #FF9441;
    font-size: 14px;
    padding: 3px 5px;
    margin-right: 3px;
    margin-top: 3px;
    border-radius: 3px;
    text-transform: uppercase;
    cursor: pointer;
    outline: none;
}

.filter-item.selected {
    background-color: #FF9441;
    color: #ffffff;
    font-weight: bold;
}
