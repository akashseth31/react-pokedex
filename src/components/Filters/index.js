import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './Filters.css';

export const FilterItem = ({ name, toggleFilters, selected }) => {
  return (
    <button className={`filter-item ${selected ? 'selected' : ''}`} onClick={() => toggleFilters(name)}>
      {name}
    </button>
  );
}

const Filters = ({ filters, toggleFilters, appliedFilters, records }) => {
  const [isOpen, toggleFilterPanel] = useState(false);
  return (
    <div className="filters">
      <button className="filter-header" onClick={() => toggleFilterPanel(!isOpen)}>
        Filter by {filters.name} (showing {records} records)
      </button>
      {isOpen &&
        <div className="filter-body">
          <div>Click on pills to toggle filters...</div>
          {
            filters.values.map(name =>
              <FilterItem
                name={name}
                toggleFilters={toggleFilters}
                key={name}
                selected={appliedFilters.includes(name)}
              />)
          }
        </div>
      }
    </div>
  )
};

Filters.propTypes = {
  toggleFilters: PropTypes.func.isRequired,
  filters: PropTypes.shape({
    name: PropTypes.string,
    values: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
  appliedFilters: PropTypes.arrayOf(PropTypes.string).isRequired,
  records: PropTypes.number.isRequired,
};


export default Filters;