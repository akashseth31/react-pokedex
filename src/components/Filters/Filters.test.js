import React from 'react';
import { mount } from 'enzyme';
import Filters, { FilterItem } from './';

const mockProps = {
  filters: {
    name: 'types',
    values: ['poison', 'grass'],
  },
  toggleFilters: jest.fn(),
  appliedFilters: ['poison'],
  records: 100,
};

describe('Filters', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Filters {...mockProps} />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render dropdown body when not open', () => {
    expect(wrapper.find('.filter-body')).toHaveLength(0);
  })

  it('should not render dropdown body when not open', () => {
    wrapper.find('.filter-header').first().simulate('click');
    expect(wrapper.find('.filter-body')).toHaveLength(1);
    expect(wrapper.find(FilterItem)).toHaveLength(mockProps.filters.values.length);
  });
});

describe('FilterItem', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<FilterItem {...mockProps} />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleFilters when clicked', () => {
    wrapper.find('button').first().simulate('click');
    expect(wrapper.props().toggleFilters).toHaveBeenCalled();
  });
});