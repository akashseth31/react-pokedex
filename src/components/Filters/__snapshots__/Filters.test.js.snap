// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem renders correctly 1`] = `ReactWrapper {}`;

exports[`Filters renders correctly 1`] = `ReactWrapper {}`;
